### ISSUE TITLE
Trim blank line around raw block with table before doing HTML parsing

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#9867]
From #9681, issue was fixed by removing the rawblock - though I understand now why there was a new slides I think 

In this parsed HTML tables 

`````markdown
```{=html}
<table>
 <thead>
  <tr>
   <th style="text-align:right;"> Sepal.Length </th>
   <th style="text-align:right;"> Sepal.Width </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;"> 5.1 </td>
   <td style="text-align:right;"> 3.5 </td>
  </tr>
</tbody>
</table>

```
`````

There is a blank line in the raw HTML block. This is what caused the insertion of the html RawBlock in our processing 
```` 
  - t: "RawBlock"
    format: "html"
    text: "\n"
````

And it seems it lead to a new slide in Powerpoint. 

Just sharing this in case another fix would be to trim blank lines around html content inside a rawblock before parsing the HTML in our processing;

_Originally posted by @cderv in https://github.com/quarto-dev/quarto-cli/issues/9681#issuecomment-2117086665_

**Comment by cscheid** [comment]
Unfortunately, I don't think we can generally remove whitespace from HTML documents, because it has layout implications. That is, we can't look at

````
```{=html}

```
````

And choose to remove that.

**Comment by cscheid** [comment]
Sorry, I should have been clearer.

I'm trying to say that, by itself, we can't simply remove a rawblock that has a single newline, because it's possible that someone added that rawblock with a newline for a reason (since it can affect HTML output).

In the context of our own processing, I do think it makes sense to _not emit_ a rawblock with a single newline in the process of splitting rawblocks that we know we want to treat.

In other words, the fix shouldn't be to add a separate filter that drops all rawblocks that only have whitespace; the fix is to change the table processing code directly.

**Comment by cderv** [comment]
> In other words, the fix shouldn't be to add a separate filter that drops all rawblocks that only have whitespace; the fix is to change the table processing code directly.

Oh perfect. We are on the same page ! 💯 agree with that, that was my plan.
